airflow/contrib/hooks/imap_hook.py

```diff
@@ -30,72 +34,105 @@ class ImapHook(BaseHook):
+
+        return self
+
+    def has_mail_attachment(self, name, check_regex=False, mail_folder='INBOX', mail_filter='All'):
         """
         Checks the mail folder for mails containing attachments with the given name.
 
         :param name: The name of the attachment that will be searched for.
         :type name: str
-        :param mail_folder: The mail folder where to look at.
-        :type mail_folder: str
         :param check_regex: Checks the name for a regular expression.
         :type check_regex: bool
+        :param mail_folder: The mail folder where to look at.
+        :type mail_folder: str
+        :param mail_filter: If set other than 'All' only specific mails will be checked.
+
+            .. seealso::
+                https://docs.python.org/3.6/library/imaplib.html#imaplib.IMAP4.search
+        :type mail_filter: str
         :returns: True if there is an attachment with the given name and False if not.
         :rtype: bool
         """
         mail_attachments = self._retrieve_mails_attachments_by_name(name,
-                                                                    mail_folder,
                                                                     check_regex,
-                                                                    latest_only=True)
+                                                                    True,
+                                                                    mail_folder,
+                                                                    mail_filter)
         return len(mail_attachments) > 0
 
     def retrieve_mail_attachments(self,
                                   name,
-                                  mail_folder='INBOX',
                                   check_regex=False,
                                   latest_only=False,
+                                  mail_folder='INBOX',
+                                  mail_filter='All',
                                   not_found_mode='raise'):
         """
         Retrieves mail's attachments in the mail folder by its name.
 
         :param name: The name of the attachment that will be downloaded.
         :type name: str
-        :param mail_folder: The mail folder where to look at.
-        :type mail_folder: str
         :param check_regex: Checks the name for a regular expression.
         :type check_regex: bool
-        :param latest_only: If set to True it will only retrieve
-                            the first matched attachment.
+        :param latest_only: If set to True it will only retrieve the first matched attachment.
         :type latest_only: bool
+        :param mail_folder: The mail folder where to look at.
+        :type mail_folder: str
+        :param mail_filter: If set other than 'All' only specific mails will be checked.
+
+            .. seealso::
+                https://docs.python.org/3.6/library/imaplib.html#imaplib.IMAP4.search
```

Comment:
I wonder if we can use this
```suggestion
                :py:func:`imaplib.imaplib.IMAP4.search`
```

(I may not have the syntax excatly right)

Reply:
Yes, that should work, have seen it somewhere - need to check for the right syntax, too :)